Pass on congestion section (#68)
ietf-wg-scone/appman: draft-ietf-scone-applicability-manageability.md

support tiered subscriber data plans (see {{Section 3.2 of I-D.ietf-scone-protocol}}).

## Interworking with Other Congestion Management Mechanisms

As stated in {{Section 3.1 of I-D.ietf-scone-protocol}},
SCONE throughput advice is not a substitute for congestion control
utilizing congestion signals such as packet loss based on transport acknowledgments, delay, or
Explicit Congestion Notification (ECN) {{RFC3168}} as also used by Low Latency, Low Loss, and Scalable

> Throughput (L4S) {{RFC9330}}. Rather, they are complementary.

Reviewer: As stated in {{Section 3.1 of I-D.ietf-scone-protocol}}, SCONE
throughput advice is not a substitute for congestion control, which
reacts to congestion signals such as packet loss, delay, and
Explicit Congestion Notification (ECN) {{RFC3168}}. Low Latency, Low
Loss, and Scalable Throughput (L4S) {{RFC9330}} also relies on ECN
for signaling. Rather, SCONE and congestion control are
complementary.

Author: ```suggestion
As stated in {{Section 3.1 of I-D.ietf-scone-protocol}},
SCONE throughput advice is not a substitute for congestion control
utilizing congestion signals such as packet loss based on transport acknowledgments, delay, or
Explicit Congestion Notification (ECN) {{RFC3168}}. Low Latency, Low
Loss, and Scalable Throughput (L4S) {{RFC9330}} also relies on ECN
for signaling. Rather, they are complementary.
```

Author: I'm okay with this change but it breaks the reading flow a bit. Is there a problem with the original sentence?

Reviewer: @mirjak agree on re-reading I'd stay with your original here. Please ignore my suggestion for lines 345 to 349.

Congestion control applies a throughput limit different from the signaled SCONE advice.
Congestion control manages the immediate dynamics of the bottleneck
link, while SCONE informs the application of the maximum rate allowed by network policy.
As such, congestion signals provide real-time
information on transient congestion for a network path, as input for congestion control that typically operates
on the time scale of a round-trip time (RTT). In contrast, SCONE throughput advice operates
over a much longer period.
Often, the throughput advice is expected to be below the
congestion limit and when the application adheres to the advice, congestion control would be
application-limited and not go into action. However, in cases of high load, congestion control would
limit the throughput below the provided advice, as the SCONE advice is only an upper limit. As such,
congestion control or a similar mechanism to react to congestion, such as a circuit breaker, is always

> needed in addition to SCONE.

Reviewer: Congestion control applies a throughput limit different from the
signaled SCONE advice. Congestion control manages the immediate
dynamics of the bottleneck link, while SCONE informs the application
of the maximum bitrate allowed by network policy. Congestion signals
provide real-time information on transient congestion for a network
path, typically operating on the time scale of a round-trip time
(RTT), whereas SCONE throughput advice operates over a much longer
period.

In the best case, the throughput advice is below the congestion
limit, and when the application adheres to the advice, congestion
control would be application-limited and not go into action.
However, in cases of high load, congestion control would limit the
throughput below the provided advice, as the SCONE advice is only an
upper limit. As such, congestion control or a similar mechanism to
react to congestion, such as a circuit breaker, is always needed in
addition to SCONE.

Author: ```suggestion
Congestion control applies a throughput limit different from the signaled SCONE advice.
Congestion control manages the immediate dynamics of the bottleneck
link, while SCONE informs the application of the maximum rate allowed by network policy.
Congestion signals provide real-time
information on transient congestion for a network path, typically operating
on the time scale of a round-trip time (RTT), whereas  SCONE throughput advice operates
over a much longer period.

When the throughput advice is below the
congestion limit and the application adheres to it, congestion control remains
application-limited and does not act. However, in cases of high load, congestion control would
limit the throughput below the provided advice, as the SCONE advice is only an upper limit. As such,
congestion control or a similar mechanism to react to congestion, such as a circuit breaker, is always
needed in addition to SCONE.
```

Author: I not sure about the "In the best case" because not sure "best" is really qualified here... but that's just a minor point

Reviewer: good point, here is what I updated the text. Let me know if this is better.

Operators can use SCONE to communicate static or dynamically adapted advice based on available
policy information, e.g. based on subscriber data, or load management policies, and
are expected to provide this information independent of
instantaneous link congestion. It is then up to the applications, such as adaptive bitrate video

> clients or bulk downloads, to utilize this advice according to their specific use cases.

Reviewer: Operators can use SCONE to communicate static or dynamically adapted
advice based on available policy information, such as subscriber
data or load management policies, independent of instantaneous link
congestion. It is then up to the applications, such as adaptive
bitrate video clients or bulk downloads, to utilize this advice
according to their specific use cases.

Author: ```suggestion
Operators can use SCONE to communicate static or dynamically adapted advice based on available
policy information, e.g. based on subscriber data, or load management policies.
Operators set this advice independent of
instantaneous link congestion. Congestion control, not SCONE, is responsible for reacting to 
congestion. It is then up to the applications, such as adaptive bitrate video
clients or bulk downloads, to utilize this advice according to their specific use cases.
```

Author: I think the original sentence was strong in explicitly saying that operators should not take congestion into account. Maybe we can find a sentence that is even more clear than that?

Reviewer: @mirjak the text you have was fine except that "are expected to" appeared as prescriptive and given previous edict of avoiding prescriptive is why I softened the language. I have now made another pass. for e.g., the new wording keeps that intent but states it directly rather than as an expectation: "Operators set this advice independent of instantaneous link congestion. Congestion control, not SCONE, is responsible for reacting to congestion.
Let me know if that works.

Network operators can deploy SCONE alongside L4S or standard ECN as two
independent network functions. Real-time congestion feedback mechanisms remain outside the SCONE domain
as SCONE advice is carried within the QUIC payload, which does not interact
with or modify ECN markings of the IP-layer ECN field.
In environments where both are present, Network operators
will benefit from ensuring that throughput advice policies and congestion control configurations are consistent

> within scoped deployments, to avoid providing conflicting feedback to applications.

Reviewer: Network operators can deploy SCONE alongside L4S or standard ECN as
two independent network functions. SCONE advice is carried within
the QUIC payload, independent of the IP-layer ECN field, and
updating the advice does not interact with or modify ECN markings.
Real-time congestion feedback mechanisms remain outside the SCONE
domain.

In environments where both are present, network operators will
benefit from ensuring that throughput advice policies and congestion
control configurations are consistent within scoped deployments, to
avoid providing conflicting feedback to applications.

Author: ```suggestion
Network operators can deploy SCONE alongside L4S or standard ECN as two
independent network functions. 
SCONE advice is carried within the QUIC payload, which does not interact
with or modify ECN markings of the IP-layer ECN field.
Real-time congestion feedback mechanisms remain outside the SCONE domain.

In environments where both are present, network operators
will benefit from ensuring that throughput advice policies and congestion control configurations are consistent
within scoped deployments, to avoid providing conflicting feedback to applications.
```

Congestion control applies a throughput limit different from the signaled SCONE advice.
Congestion control manages the immediate dynamics of the bottleneck
link, while SCONE informs the application of the maximum rate allowed by network policy.
Congestion signals provide real-time
information on transient congestion for a network path, typically operating
on the time scale of a round-trip time (RTT), whereas  SCONE throughput advice operates
over a much longer period.

When the throughput advice is below the
congestion limit and the application adheres to it, congestion control remains
application-limited and does not act. However, in cases of high load, congestion control would
limit the throughput below the provided advice, as the SCONE advice is only an upper limit. As such,
congestion control or a similar mechanism to react to congestion, such as a circuit breaker, is always
needed in addition to SCONE.

Operators can use SCONE to communicate static or dynamically adapted advice based on available
policy information, e.g. based on subscriber data, or load management policies.
Operators set this advice independent of

> instantaneous link congestion. It is then up to the applications, such as adaptive bitrate video

Author: or maybe we merge both sentences like this:

```suggestion
Operators set this advice independent of
instantaneous link congestion and do not need to coordinate throughput advise policy
with congestion control. It is then up to the applications, such as adaptive bitrate video
```

Reviewer: Good idea, cleaner as one sentence. Fixed a couple of typos (advise->advice, and matched "policies" plural used elsewhere in the doc).

```suggestion
Operators set this advice independent of
instantaneous link congestion and do not need to coordinate throughput advice policies with
congestion control. It is then up to the applications, such as adaptive bitrate video
```

Author: Thanks! Will merge!

> 
> Operators can use SCONE to communicate static or dynamically adapted advice based on available
> policy information, e.g. based on subscriber data, or load management policies.
> Operators set this advice independent of
> instantaneous link congestion. It is then up to the applications, such as adaptive bitrate video
> clients or bulk downloads, to utilize this advice according to their specific use cases.

Reviewer: Agreed, good catch -- moved the sentence up to replace the one you flagged, and removed the now-empty closing paragraph in the other thread.

```suggestion
Operators can use SCONE to communicate static or dynamically adapted advice based on available
policy information, e.g. based on subscriber data, or load management policies.
SCONE and congestion control operate independently, so network operators do not need to
coordinate throughput advice policies with congestion control. It is then up to the applications,
such as adaptive bitrate video clients or bulk downloads, to utilize this advice according to
their specific use cases.
```

Network operators can deploy SCONE alongside L4S or standard ECN as two

> independent network functions. 

Author: ```suggestion
independent network functions.
```

Congestion control manages the immediate dynamics of the bottleneck
link, while SCONE informs the application of the maximum rate allowed by network policy.
Congestion signals provide real-time
information on transient congestion for a network path, typically operating
on the time scale of a round-trip time (RTT), whereas  SCONE throughput advice operates
over a much longer period.

When the throughput advice is below the
congestion limit and the application adheres to it, congestion control remains
application-limited and does not act. However, in cases of high load, congestion control would
limit the throughput below the provided advice, as the SCONE advice is only an upper limit. As such,
congestion control or a similar mechanism to react to congestion, such as a circuit breaker, is always
needed in addition to SCONE.

Operators can use SCONE to communicate static or dynamically adapted advice based on available
policy information, e.g. based on subscriber data, or load management policies.
Operators set this advice independent of
instantaneous link congestion. Congestion control, not SCONE, is responsible for reacting to 

> congestion. It is then up to the applications, such as adaptive bitrate video

Author: This new sentence on cc being responsible for congestion is a bit redundant. The whole purpose of the previous paragraphs is to explain that. I think it would be better to only focus in this paragraph on setting the scone advice (independent of cc).

Reviewer: @mirjak, Agreed, cut. Sentence is redundant with the prior paragraph's "congestion control... is always needed in addition to SCONE." Suggestion below removes it.

```suggestion
Operators can use SCONE to communicate static or dynamically adapted advice based on available
policy information, e.g. based on subscriber data, or load management policies.
Operators set this advice independent of
instantaneous link congestion. It is then up to the applications, such as adaptive bitrate video
clients or bulk downloads, to utilize this advice according to their specific use cases.
```

Author: ```suggestion
instantaneous link congestion. It is then up to the applications, such as adaptive bitrate video
```

clients or bulk downloads, to utilize this advice according to their specific use cases.

Network operators can deploy SCONE alongside L4S or standard ECN as two
independent network functions. 
SCONE advice is carried within the QUIC payload, which does not interact
with or modify ECN markings of the IP-layer ECN field.
Real-time congestion feedback mechanisms remain outside the SCONE domain.

In environments where both are present, network operators
will benefit from ensuring that throughput advice policies and congestion control configurations are consistent

> within scoped deployments, to avoid providing conflicting feedback to applications.

Author: I merged this, so we have a clean PR, but really don't know what is meant here? How should an operator keep these configuration consistent? The whole point is these are two independent mechanisms and there is nothing an operator must or should do to align them.

Reviewer: @mirjak, Fair point, no concrete alignment action to point to. Rewrote to state that plainly.

```suggestion
In environments where both are present, SCONE and congestion control operate independently,
so network operators do not need to coordinate throughput advice policies with congestion
control.
```

Author: This is good but also now a bit redundant to the sentence two paragraphs up:

> Operators set this advice independent of instantaneous link congestion.

I actually like this new sentence here more because it is clear about the guidance provided. But maybe I move it up and replace the sentence above?

Reviewer: @mirjak, Moving this up per your suggestion above -- see the new comment on the earlier paragraph.

```suggestion
```
